service-loadbalancer/service_loadbalancer.go

```diff
@@ -635,7 +651,11 @@ func main() {
 		glog.Fatalf("unexpected error: %v", err)
 	}
 	if !specified {
-		namespace = "default"
+		namespace = api.NamespaceDefault
```

Comment:
to make this ordering clearer (i.e that allNamespaces always wins) please make it an if/elseif/else where in the "specified" case you just log the namespace.